sdk/clientcore/core/src/main/java/io/clientcore/core/http/client/HttpClient.java

```diff
@@ -22,6 +24,30 @@ public interface HttpClient {
      */
     Response<BinaryData> send(HttpRequest request);
 
+    /**
+     * Sends the provided request asynchronously with contextual information.
+     * <p>
+     * The default implementation of this method wraps the synchronous {@link #send(HttpRequest)} method with
+     * {@link CompletableFuture#completedFuture(Object)} if the request is successful. If the request completes
+     * exceptionally a {@link CompletableFuture} failed with {@link CompletableFuture#completeExceptionally(Throwable)}
+     * is returned instead.
+     *
+     * @param request The HTTP request to send.
+     * @return A {@link CompletableFuture} that completes with the response, or fails with an exception if the request
+     * fails.
+     */
+    default CompletableFuture<Response<BinaryData>> sendAsync(HttpRequest request) {
```

Comment:
Given that we have `httpClient` method on the client builders, having sync and async on a single `HttpClient` interface makes the user code simpler compared to separate HttpClient and AsyncHttpClient interfaces.